interview-questions.md

```diff
@@ -8,42 +8,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: The difference between and object oriented programming (OOP) is that functional programming relies on functions to create logic and manage data. In object oriented programming the data is organized as object, which themselves have their own "functionality" in the form of methods. In OOP object can be classes which enable to create subclasses (child classes) that can both inherit and further specialize that data block as required.
 
-Researched answer:
+Researched answer: The principles of Object oriented programming are:  
 
-1. What is the difference between a Float and an Integer in Ruby?
+- Encapsulation, where you contain logic inside objects.
+- Abstraction, which allow only pertinent code to be shown while hiding unnecessary code.
+- Inheritance, which I meantioned before in where a parent class creates a blueprint for child classes.
+- Polymorphism, where child classes, while inheriting properties of the parent, can also other properties required for specialization.
 
-Your answer:
+2. What is the difference between a Float and an Integer in Ruby?
 
-Researched answer:
+Your answer: The difference between a float and an integer in Ruby is that a float requires a higher degree of precision, so a float variable requires more memory allocation than an integer.
 
-1. Ruby has an implicit return. What does that mean?
+Researched answer: in my researched answer I found the same thing in that a float provides a higher degree of preceision but an integer requires less memory from the program so it is faster to process, so it is worthwile using an integer when a high degree of accuracy is not required. 
 
```

Comment:
Would be good to mention how an Integer is a whole number while a Float is a fraction/decimal as well